packages/peregrine/lib/talons/CartPage/PriceAdjustments/useCouponCode.js

```diff
@@ -92,16 +91,15 @@ export const useCouponCode = props => {
         setIsCartUpdating
     ]);
 
-    const derivedErrorMessage = useMemo(
-        () => deriveErrorMessage([applyError, removeCouponError]),
-        [applyError, removeCouponError]
-    );
+    const errors = useMemo(() => new Map(), []);
+    errors.set('applyCouponMutation', applyError);
+    errors.set('removeCouponMutation', removeCouponError);
+    errors.set('getAppliedCouponsQuery', fetchError);
```

Comment:
I believe this would create a new Map every loop; would this be better?

```suggestion
    const errors = useMemo(() => new Map([
        ['applyCouponMutation', applyError],
        ['removeCouponMutation', removeCouponError],
        ['getAppliedCouponsQuery', fetchError]
    ]), [applyError, fetchError, removeCouponError]);
```

Reply:
We need to be able to create the map but set and unset each error instance individually. I think an effect for each might be required, as your suggestion would recreate the map every time an error occurred or the query response changed.

Reply:
I also don't think this will create a new error on loop - it should only create once on mount since there are no dependencies.

Reply:
Discussed with @jimbo memoizing a single map on init will not work if folks want to use the map's identity to do some work in the component. Better to recreate the map when the properties change.